Gem/Code/Source/Components/UI/HUDComponent.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Components/UI/HUDComponent.h>
+#include <Source/Components/NetworkMatchComponent.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+
+#include <LyShine/Bus/UiCanvasBus.h>
+#include <LyShine/Bus/UiTextBus.h>
+
+namespace MultiplayerSample
+{
+    void HUDComponent::Activate()
+    {
+        UiCanvasRefBus::EventResult(m_uiCanvasId, GetEntityId(), &UiCanvasRefBus::Events::GetCanvas);
+        if (!m_uiCanvasId.IsValid())
+        {
+            UiCanvasAssetRefNotificationBus::Handler::BusConnect(GetEntityId());
+        }
+
+        NetworkMatchComponent* netMatchComponent = GetEntity()->FindComponent<NetworkMatchComponent>();
+
+        SetRoundNumberText(netMatchComponent->GetRoundNumber());
+        m_roundNumberHandler = AZ::EventHandler<uint16_t>([this](uint16_t value) { this->SetRoundNumberText(value); });
+        netMatchComponent->RoundNumberAddEvent(m_roundNumberHandler);
+
+        m_roundTimerHandler = AZ::EventHandler<float>([this](float value) { this->SetRoundTimerText(value); });
+        netMatchComponent->RoundTimeAddEvent(m_roundTimerHandler);
+    }
+
+    void HUDComponent::Deactivate()
+    {
+        UiCanvasAssetRefNotificationBus::Handler::BusDisconnect(GetEntityId());
+    }
+
+    void HUDComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<HUDComponent, AZ::Component>()
+                ->Version(1)
+                ->Field("RoundNumberText", &HUDComponent::m_roundNumberText)
+                ->Field("RoundNumberId", &HUDComponent::m_roundNumberId)
+                ->Field("RoundTimerText", &HUDComponent::m_roundTimerText)
+                ->Field("RoundTimerId", &HUDComponent::m_roundTimerId)
+                ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<HUDComponent>("HUDComponent", "Helper component for setting up Multiplayer Sample's HUD")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Multiplayer Sample UI")
+                    ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/Component_Placeholder.svg")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &HUDComponent::m_roundNumberId, "Round Number UIElement Id", "The element id of the textbox you want to change. (Find the id inside the ui canvas)")
```

Comment:
Same comment here you don't need to find the id inside the canvas if you place this component as a CanvasUI component, then it can live on the canvas directly and you can drag and drop elements into these entity variables.

And then you can directly call: (assuming m_roundTimerId is AZ::EntityId)
```
UiTextBus::Event(m_roundTimerId, &UiTextBus::Events::SetText, m_roundNumberText);
```